eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/AbstractLazyIterable.java

```diff
@@ -127,6 +128,25 @@ public T getLast()
     }
 
     @Override
+    public T getOnly()
+    {
+        Iterator<T> iterator = this.iterator();
+
+        if (!iterator.hasNext())
+        {
+            throw new IllegalStateException("Iterable is empty.");
+        }
+
+        T result = iterator.next();
+        if (iterator.hasNext())
+        {
+            throw new IllegalStateException("Iterable has multiple values.");
+        }
+
+        return result;
```

Comment:
This is the correct implementation for AbstractLazyIterable, where we don't want to call size(). Besides here, we should try to eliminate all usages of iterators.